policy/policy.go

```diff
@@ -38,57 +38,75 @@ const (
 )
 
 var (
-	emptyPolicy            Policy
-	emptyVersionedPolicies VersionedPolicies
+	// EmptyPolicy represents an empty policy
+	EmptyPolicy Policy
 
-	// defaultPolicies are the default policies
+	// EmptyVersionedPolicies represents an empty VersionPolicies
+	EmptyVersionedPolicies VersionedPolicies
+
+	// DefaultPolicies are the default policies
 	// TODO(xichen): possibly make this dynamically configurable in the future
-	defaultPolicies = []Policy{
-		{
-			Resolution: Resolution{Window: 10 * time.Second, Precision: xtime.Second},
-			Retention:  Retention(2 * 24 * time.Hour),
+	DefaultPolicies = []Policy{
+		&policy{
+			resolution: Resolution{Window: 10 * time.Second, Precision: xtime.Second},
+			retention:  Retention(2 * 24 * time.Hour),
 		},
-		{
-			Resolution: Resolution{Window: time.Minute, Precision: xtime.Minute},
-			Retention:  Retention(30 * 24 * time.Hour),
+		&policy{
+			resolution: Resolution{Window: time.Minute, Precision: xtime.Minute},
+			retention:  Retention(30 * 24 * time.Hour),
 		},
 	}
 )
 
 // Policy represents the resolution and retention period metric datapoints
 // are stored at
-type Policy struct {
+type Policy interface {
```

Comment:
hm any reason why we need to make it an interface? I intentionally kept it a struct type so they are always allocated on the stack and we don't have GC sweeping them all the time